src/networking.c

```diff
@@ -2758,18 +2758,17 @@ NULL
     }
 }
 
-/* HELLO <protocol-version> [AUTH <user> <password>] [SETNAME <name>] */
+/* HELLO [protocol-version] [AUTH <user> <password>] [SETNAME <name>] */
 void helloCommand(client *c) {
-    long long ver;
+    long long ver = 0;
 
-    if (getLongLongFromObject(c->argv[1],&ver) != C_OK ||
-        ver < 2 || ver > 3)
-    {
+    if (c->argc >= 2 && getLongLongFromObject(c->argv[1],&ver) == C_OK &&
+            (ver < 2 || ver > 3)) {
         addReplyError(c,"-NOPROTO unsupported protocol version");
         return;
     }
 
-    for (int j = 2; j < c->argc; j++) {
+    for (int j = !ver ? 1 : 2; j < c->argc; j++) {
```

Comment:
this is a bit more readable IMHO.
we can even take it a step further and eliminate the use of `j`
```suggestion
    for (int j = next_arg; j < c->argc; j++) {
```